libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Selectors/RandomSelector.cs

```diff
@@ -75,9 +77,18 @@ public override Task<IReadOnlyList<OnCondition>> SelectAsync(ActionContext conte
             if (candidates.Count > 0)
             {
                 int selection;
-                lock (_objectLock)
+                var (value, error) = Expression.Parse("Conversation.TestOptions").TryEvaluate<IDictionary<string, object>>(context.State);
+                if (error == null && value != null)
```

Comment:
I don't think test options is an expression.  Conversation.testOptions is just plain old memory like everything else.  Someone could have an expression in SetProperty to compute this, but the memory itself would not be an expression.